### PR TITLE
Vulcanization of R4Medication

### PR DESCRIPTION
**Pull request by bcastillomartel-va**

### CODE REVIEW

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medication/MedicationSamples.java [hunk 1]
```diff
@@ -15,7 +16,23 @@
 
 @UtilityClass
 public class MedicationSamples {
-  static final ObjectMapper MAPPER = JacksonConfig.createMapper();
+
+  public static ResourceIdentity id(String cdwId) {
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medication/MedicationSamples.java:20]: This method (and `registration()` below) are unnecessary. Take a look at `TestIdentityService.registerMockIdentities()`

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/R4MedicationController.java [hunk 2]
```diff
@@ -1,114 +1,126 @@
...
   @GetMapping(value = "/raw/{publicId}")
   String readRaw(@PathVariable("publicId") String publicId, HttpServletResponse response) {
-    IncludesIcnMajig.addHeaderForNoPatients(response);
-    return findById(publicId).payload();
+    return vulcanizedReader().readRaw(publicId, response);
+  }
+
+  /** Search Support. */
+  @GetMapping
+  public Medication.Bundle search(HttpServletRequest request) {
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/medication/R4MedicationController.java:87]: You can make this package-protected and remove the redundant javadoc. (Similarly `read()` can also be protected)

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medication/MedicationSamples.java [hunk 3]
```diff
@@ -24,7 +41,7 @@ static MedicationEntity entity(DatamartMedication dm) {
       return MedicationEntity.builder()
           .cdwIdNumber(CompositeCdwId.fromCdwId(dm.cdwId()).cdwIdNumber())
           .cdwIdResourceCode(CompositeCdwId.fromCdwId(dm.cdwId()).cdwIdResourceCode())
-          .payload(MAPPER.writeValueAsString(dm))
+          .payload(json(dm))
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medication/MedicationSamples.java:44]: remove `static` from `entity()` method to be consistent with other samples

Comment (bcastillomartel-va) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/medication/MedicationSamples.java:30]: this broke the tests in `DSTU2MedicationControllerTest` , so it's still in progress

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**